### DISCUSSION TITLE
Ingesting form data into a bean without annotations

### DISCUSSION THREAD

**Discussion by lhgravendeel** [quarkusio/quarkus#35351] [Q&A]
I'm looking for a way to ingest regular posted form data into a simple java class with no annotations on it.

The incoming request is sending HTTP POST data (`Content-Type: application/x-www-form-urlencoded`):
```
firstName=John&lastName=Doe
```

If the receiving class did have annotations, it would look like this with quarkus-resteasy-reactive and Quarkus 3.2.3:
```
public class DummyForm {
  @RestForm
  public String firstName;
  @RestForm
  public String lastName;
}
```

The receiving method then works like this:
```
@POST
@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
public String formTestResult(@BeanParam DummyForm dummyForm) {
  return "ok";
}
```

Up to this part, it all works fine, but I'm trying to figure out how to make this work with a class similar to DummyForm - without annotations. For example, I would like to ingest this form data as this `PlainDummyForm`:
```
public class PlainDummyForm {
  public String firstName;
  public String lastName;
}
```

If I try this out with a controller method, Quarkus will return a status code 415 (Unsupported Media Type) :
```
@POST
@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
public String formTestResult(PlainDummyForm dummyForm) {
  return "ok";
}
```

I also tried out `public String formTestResult(@BeanParam PlainDummyForm dummyForm)`, but this triggers an exception:
```Caused by: jakarta.enterprise.inject.spi.DeploymentException: No annotations found on fields at 'nl.gravendeeldesign.studio.forms.DummyForm'. Annotations like `@QueryParam` should be used in fields, not in methods```

Just for comparison - if I were sending JSON data rather than form data, the approach with `PlainDummyForm` and no annotations seems to work well if I'm using quarkus-resteasy-reactive-jackson.

Is there any way to make the approach with `PlainDummyForm` and no annotations work for form data as well?

**Comment by bulldog98** [comment]
\cc @maxandersen not sure which label would match here
